swap_meet/vendor.py

```diff
@@ -0,0 +1,155 @@
+            self.inventory = inventory 
+
+    def add(self, item): 
+        '''
+        Adds one item to the vendor's inventory
+        Parameters: 
+            one item object of class Item 
+        Returns: 
+            one item object of class Item 
+        '''
+        self.inventory.append(item)
+        return item
+        
+    def remove(self, item): 
+        '''
+        Removes one item that exists in the vendor's inventory and returns the item if found
+        Parameters: 
+            item
+        Returns: 
+            item or False
+        '''
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item
+        else: 
+            return False
+
+# ---- Wave 2 ----- #
+    def get_by_category(self, category): 
+        '''
+        Creates a list of items in a given category
+        Parameters: 
+            category (str)
+        Returns: 
+            list of items
+        '''
+        items_list = []
+        for item in self.inventory: 
+            if category == item.category: 
+                items_list.append(item)
+        return items_list 
+
+# ---- Wave 3 ----- #
+    def swap_items(self, friend, my_item, their_item):
+        '''
+        Swaps items of a given category between 2 vendors
+        Parameters: 
+            friend (Vendor), my_item (Item), their_item (Item)
+        Returns: 
+            True if exchange occured or False if exchange didn't occur
+        '''
+        swapped = False
+        if my_item in self.inventory and their_item in friend.inventory:
+            self.inventory.remove(my_item)
+            friend.inventory.append(my_item)
+            self.inventory.append(their_item)
+            friend.inventory.remove(their_item)
+            swapped = True
+        return swapped
+
+# ---- Wave 4 ----- #
+    def swap_first_item(self, friend): 
+        '''
+        Swaps first items in inventories of 2 vendors. 
+        Parameters: 
+            friend (Vendor)
+        Returns: 
+            True if exchange occured or False if exchange didn't occur
+        '''
+        if not self.inventory or not friend.inventory: 
+            return False 
+        return self.swap_items(friend, self.inventory[0], friend.inventory[0])
+
+# ---- Wave 6 ----- #
+    def get_best_by_category(self, category): 
+        '''
+        Gets items in given category and finds the item in the best condition. 
+        Returns None if no item in given category was found.
+        Parameters: 
+            category (str)
+        Returns: 
+            best_item (Item) or None
+        '''
+        category_items = self.get_by_category(category)
+        items_conditions = []
+        best_item = None
+        for item in category_items: 
+            items_conditions.append(item.condition)
+            best_condition = max(items_conditions)
+            if best_condition == item.condition:  
+                best_item = item
+        return best_item
```

Comment:
so this works! But there's a lot of for loops going on here. 1 from `get_by_category`, 1 in the code, and at least 1 in `max` method under the hood. 

```suggestion
        category_items = self.get_by_category(category)
        best_item = category_items[0]
        for item in category_items: 
            if item.condition > best_item.condition:
                best_item = item
        return best_item
```